### ISSUE TITLE
No accessible label or state on 

### ISSUE THREAD

**Issue by cmcmenamin** [mdn/yari#8015]
### Summary

In the mobile view, there is a button you can use to expand and collapse the sidebar.

This sidebar button has no accessible label, or expanded state, which means screen reader users just head 'button', without hearing what the button is for, and whether it's expanded or collapsed.

### URL

https://developer.mozilla.org/en-US/docs/Web/HTML

### Reproduction steps

1. Go into mobile view
2. Use a screen reader to announce the button
3. Hear that the button has no accessible name
4. Alternatively, Inspect the button with Chrome DevTools and check the 'Name' property in the Accessibility tab

### Expected behavior

Screen reader should announce "Sidebar expanded, button"

### Actual behavior

"Button".

### Device

Smartphone

### Browser

Chrome

### Browser version

Extended Support Release

### Operating system

iOS

### Screenshot

![button](https://user-images.githubusercontent.com/112744591/213918287-9e3b772f-70dc-41a5-bca3-264bf9a5ba6a.png)

### Validations

- [X] I have read the [Community Participation Guidelines](https://www.mozilla.org/en-US/about/governance/policies/participation/).
- [X] I have verified that there isn't [already an issue](https://github.com/mdn/yari/issues) that reports the same bug to avoid creating a duplicate.
- [X] I have checked that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/orgs/mdn/discussions/categories/platform).
